docs/deployment/updating_the_stack.rst

```diff
@@ -0,0 +1,96 @@
+.. _deployment-updating-the-stack:
+
+==================
+Updating the Stack
+==================
+
+The entire COPS system is deployed using Docker swarm. Docker swarm provides a
+``docker stack`` command that will deploy a set of services based on a docker-compose
+file. The stack can also be updated when there are changes. The deployment process
+is currently done manually but is fairly straightforward.
+
+The steps at a high level are:
+
+1. Establish an SSH tunnel from your local computer to one of the swarm management
+nodes.
+2. Run the ``./script/build-push.sh`` script to build and push images to dockerhub.
+3. Run the ``./script/deploy.sh`` script to update the stack.
+
+The more granular details of the deployment are explained below.
+
+Install docker-auto-labels
+==========================
+
+The docker-auto-labels package is used to ensure the proper labels are applied to the
+docker swarm nodes. For example, the database should only be running on ``cc1.cnx.org``.
+This is done by applying a label to that node and adding a constraint to the
+docker-compose file.
+
+Install the docker-auto-labels package:
+
+.. code-block:: bash
+
+   pip install docker-auto-labels
+
+
+Setup the SSH tunnel
+====================
+
+Open a fresh terminal window. Keep it open until the end of the deployment process.
+
+Run the following command to establish an SSH tunnel to a manager node:
+
+.. code-block:: bash
+
+   ssh -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -NL localhost:2377:/var/run/docker.sock <user>@cc1.cnx.org
+
+This command doesn't produce any output unless there is an error. No other commands
+will be typed into this window.
+
+Setup terminal window for communicating with docker swarm manager node
+======================================================================
+
+Open another fresh terminal window. Keep it open until the end of the process.
+
+Configure docker in that terminal to use the remote host established before:
+
+.. code-block:: bash
+
+   export DOCKER_HOST="http://localhost:2377"
```

Comment:
`export DOCKER_HOST="http://localhost:2377"` produces error, should be
export DOCKER_HOST="localhost:2377"

Reply:
This has been resolved